docs/drafts/derive-serviceaccount.md

```diff
@@ -0,0 +1,578 @@
+  resources: [clusterextensions/finalizers]
+  verbs: [update]
+  resourceNames: [<cluster-extension-name>]
+```
+
+CRUD for Custom Resource Definitions
+
+```yaml
+- apiGroups: [apiextensions.k8s.io]
+  resources: [customresourcedefinitions]
+  verbs: [create, list, watch]
+
+- apiGroups: [apiextensions.k8s.io]
+  resources: [customresourcedefinitions]
+  verbs: [get, update, patch, delete]
+  resourceNames: [<crd name 1>, ..., <crd name n>]
+```
+
+CRUD for Cluster Roles
+
+```yaml
+- apiGroups: [rbac.authorization.k8s.io]
+  resources: [clusterroles]
+  verbs: [create, list, watch]
+
+- apiGroups: [rbac.authorization.k8s.io]
+  resources: [clusterroles]
+  verbs: [get, update, patch, delete]
+  resourceNames: [<generated cluster role 1>, ..., <generated cluster role n>, <manifest cluster role name 1>, ..., <manifest cluster role name n>]
+```
+
+CRUD for Cluster Role Bindings
+
+```yaml
+- apiGroups: [rbac.authorization.k8s.io]
+  resources: [clusterrolebindings]
+  verbs: [create, list, watch]
+
+- apiGroups: [rbac.authorization.k8s.io]
+  resources: [clusterrolebindings]
+  verbs: [get, update, patch, delete]
+  resourceNames: [<generated cluster role 1>, ..., <generated cluster role n>, <manifest cluster role binding name 1>, ..., <manifest cluster role binding name n>]
+```
+
+CRUD for managing deployments
+
+```yaml
+- apiGroups: [apps]
+  resources: [deployments]
+  verbs: [create, list, watch]
+
+- apiGroups: [apps]
+  resources: [deployments]
+  verbs: [get, update, patch, delete]
+  resourceNames: [argocd-operator-controller-manager]
+```
+
+CRUD for service accounts used by the deployment
+ 
+```yaml
+- apiGroups: [""]
+  resources: [serviceaccounts]
+  verbs: [create, list, watch]
+
+- apiGroups: [""]
+  resources: [serviceaccounts]
+  verbs: [get, update, patch, delete]
+  resourceNames: [argocd-operator-controller-manager]
+```
+
+### Creation of ClusterRoleBinding using the cluster-admin ClusterRole in non-production environments
+
+Below is an example ClusterRoleBinding using the cluster-admin ClusterRole for non-production environments.
+
+```yaml
+# Create ClusterRole
+kubectl apply -f - <<EOF
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: my-cluster-extension-installer-role-binding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: cluster-admin
+subjects:
+- kind: ServiceAccount
+  name: my-cluster-extension-service-account
+  namespace: my-cluster-extension-namespace
+EOF
+```
+
+Use the below on a Kind cluster to assign cluster-admin privileges to your cluster extension
+
+```sh
+kubectl create clusterrolebinding my-cluster-extension-installer-role-binding \
+  --clusterrole=cluster-admin \
+  --serviceaccount=my-cluster-extension-namespace:my-cluster-installer-service-account
+```
+
+### Example ClusterExtension with RBAC
+
+Below is an example of the ArgoCD installer with the necessary RBAC to deploy the ArgoCD ClusterExtension.
+
+??? example
```

Comment:
You can see the current commit's rendering here: https://deploy-preview-1238--olmv1.netlify.app/drafts/derive-serviceaccount/

The example block isn't quite right yet. The entire code block needs to be indented under the ??? tag. You can see an example here: https://github.com/operator-framework/operator-controller/blob/34821fa76280945f263d72e2cd118821c8554f91/docs/refs/crd-upgrade-safety.md?plain=1#L136

Reply:
![Uploading Screenshot 2024-09-20 at 12.11.33 PM.png…]()

Reply:
This block is not rendered properly. This needs to be fixed.

Reply:
it renders correctly in the netlify deploy view: https://deploy-preview-1238--olmv1.netlify.app/drafts/derive-serviceaccount/